dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -739,148 +768,213 @@ public String getWorkflowDag(@ApiParam(value = "workflowId", required = true) @P
+     * */
+    private WorkflowVersion getWorkflowVersion(Workflow workflow, Long workflowVersionId){
+        Set<WorkflowVersion> workflowVersions = workflow.getVersions();
+        WorkflowVersion workflowVersion = null;
+
+        for(WorkflowVersion wv : workflowVersions) {
+            if(wv.getId() == workflowVersionId) {
+                workflowVersion = wv;
+                break;
             }
+        }
+
+        return workflowVersion;
+    }
+
+    /**
+     * This method will find the main descriptor file based on the workflow version passed in the parameter
+     * @param workflowVersion
+     * @return mainDescriptor
+     * */
+    private SourceFile getMainDescriptorFile(WorkflowVersion workflowVersion){
 
-            // set up JSON for DAG (edges and nodes)
-            return setupJSON(nodePairs);
+        SourceFile mainDescriptor = null;
+        for (SourceFile sourceFile : workflowVersion.getSourceFiles()) {
+            if (sourceFile.getPath().equals(workflowVersion.getWorkflowPath())) {
+                mainDescriptor = sourceFile;
+                break;
+            }
         }
-        return null;
+
+        return mainDescriptor;
     }
 
     /**
-     * This method will process the DAG for WDL workflow and return pair of task id and dokerpull link
+     * This method will get the content for tool tab with descriptor type = WDL
+     * It will then call another method to transform the content into JSON string and return
      * @param tempMainDescriptor
-     * @return ArrayList
-     */
-    public ArrayList<Pair<String, String>> getWDLDAG(File tempMainDescriptor){
+     * @return String
+     * */
+    private String getContentWDL(File tempMainDescriptor, String type) {
         Bridge bridge = new Bridge();
+        Map<String, Seq> callToTask = (LinkedHashMap)bridge.getCallsAndDocker(tempMainDescriptor);
+        Map<String, Pair<String, String>> taskContent = new HashMap<>();
         ArrayList<Pair<String, String>> nodePairs = new ArrayList<>();
-        // TODO : Currently evaluates scatters last, while loops don't work.  This needs to be fixed in Bridge.scala.
-        Map<String, Seq> callToTask = (LinkedHashMap)bridge.getCallsAndDocker(tempMainDescriptor); // Should be in correct order
-        // TODO : Currently only grabs first from a possible list (implement multiple containers in the future)
+        String result = null;
+
         for (Map.Entry<String, Seq> entry : callToTask.entrySet()) {
-            if (entry.getValue() != null){
-                nodePairs.add(new MutablePair<>(entry.getKey(), getURLFromEntry(entry.getValue().head().toString())));
-            } else{
-                nodePairs.add(new MutablePair<>(entry.getKey(), ""));
+            if(type.equals("tools")){
+                if (entry.getValue() != null){
+                    taskContent.put(entry.getKey(), new MutablePair<>(entry.getValue().head().toString(), getURLFromEntry(entry.getValue().head().toString())));
```

Comment:
extract `entry.getKey()', 'entry.getValue()' and 'entry.getValue().head()' to variables and give them a good name to help explain what they are